### PR TITLE
fix(): Deleted default sales channel should be prevented

### PR DESCRIPTION
**Pull request by adrien2p**
FIXES CMRC-722

**What**
- It should not be allowed to delete a default sales channel
- The admin does not allow to delete a sales channel use as the default for the store

### CODE REVIEW

**Comment by adrien2p** [comment]
Add tooltip once this [pr](https://github.com/medusajs/medusa/pull/10234) is merged

File: packages/admin/dashboard/src/i18n/translations/en.json [hunk 1]
```diff
@@ -2308,6 +2308,9 @@
       "update": "Sales channel updated successfully",
       "delete": "Sales channel deleted successfully"
     },
+    "tooltip": {
+      "cannotDeleteDefault": "Cannot delete default sales channel"
+    },
```
Comment (adrien2p) [inline packages/admin/dashboard/src/i18n/translations/en.json:2339]: @kasperkristensen could you validate that please?

**Comment by adrien2p** [comment]
@olivermrbl should we do something about this bad boy?

**Comment by olivermrbl** [comment]
Will have a look later today ✅

**Comment by adrien2p** [comment]
> Will have a look later today ✅

any luck on this? :D

**Review by olivermrbl — APPROVED** [review]
LGTM, will let @kasperkristensen have a look at the admin-related changed :)

**Review by fPolic — APPROVED** [review]
Looks good!

Should we disallow disabling the default channel maybe as well?

**Comment by adrien2p** [comment]
> Looks good!
> 
> Should we disallow disabling the default channel maybe as well?

that's a good question, should I add it as part of this pr?

**Comment by olivermrbl** [comment]
Think we can merge this as is?

**Pull request merged**